packages/frontend/editor-ui/src/composables/usePushConnection/handlers/executionFinished.ts

```diff
@@ -0,0 +1,496 @@
+import { getEasyAiWorkflowJson } from '@/utils/easyAiWorkflowUtils';
+import { clearPopupWindowState, hasTrimmedData, hasTrimmedItem } from '@/utils/executionUtils';
+import { useWorkflowsStore } from '@/stores/workflows.store';
+import { useSettingsStore } from '@/stores/settings.store';
+import { useWorkflowHelpers } from '@/composables/useWorkflowHelpers';
+import { useTelemetry } from '@/composables/useTelemetry';
+import { parse } from 'flatted';
+import { useToast } from '@/composables/useToast';
+import type { useRouter } from 'vue-router';
+import { useI18n } from '@/composables/useI18n';
+import { TelemetryHelpers } from 'n8n-workflow';
+import type {
+	IWorkflowBase,
+	NodeError,
+	NodeOperationError,
+	SubworkflowOperationError,
+	ExpressionError,
+	IDataObject,
+	IRunExecutionData,
+} from 'n8n-workflow';
+import { codeNodeEditorEventBus, globalLinkActionsEventBus } from '@/event-bus';
+import { h } from 'vue';
+import NodeExecutionErrorMessage from '@/components/NodeExecutionErrorMessage.vue';
+import { getTriggerNodeServiceName } from '@/utils/nodeTypesUtils';
+import { useExternalHooks } from '@/composables/useExternalHooks';
+import { useNodeHelpers } from '@/composables/useNodeHelpers';
+import { useNodeTypesStore } from '@/stores/nodeTypes.store';
+
+export type SimplifiedExecution = Pick<
+	IExecutionResponse,
+	'workflowId' | 'data' | 'status' | 'startedAt' | 'stoppedAt'
+>;
+
+/**
+ * Handles the 'executionFinished' event, which happens when a workflow execution is finished.
+ */
+export async function executionFinished(
+	{ data }: ExecutionFinished,
+	options: { router: ReturnType<typeof useRouter> },
+) {
+	const workflowsStore = useWorkflowsStore();
+	const uiStore = useUIStore();
+
+	// No workflow is actively running, therefore we ignore this event
+	if (typeof workflowsStore.activeExecutionId === 'undefined') {
+		return;
+	}
+
+	const telemetry = useTelemetry();
+
+	clearPopupWindowState();
+
+	const workflow = workflowsStore.getWorkflowById(data.workflowId);
+	if (workflow?.meta?.templateId) {
+		const easyAiWorkflowJson = getEasyAiWorkflowJson();
+		const isEasyAIWorkflow = workflow.meta.templateId === easyAiWorkflowJson.meta.templateId;
+		if (isEasyAIWorkflow) {
+			telemetry.track(
+				'User executed test AI workflow',
+				{
+					status: data.status,
+				},
+				{ withPostHog: true },
+			);
+		}
+	}
+
+	uiStore.setProcessingExecutionResults(true);
+
+	let successToastAlreadyShown = false;
+	let execution: SimplifiedExecution | undefined;
+	if (data.rawData) {
+		const { workflowId, status, rawData } = data;
+
+		execution = {
+			workflowId,
+			data: parse(rawData),
+			status,
+			startedAt: workflowsStore.workflowExecutionData?.startedAt ?? new Date(),
+			stoppedAt: new Date(),
+		};
+	} else {
+		if (data.status === 'success') {
+			handleExecutionFinishedSuccessfully(data.workflowId, options);
+			successToastAlreadyShown = true;
+		}
+
+		execution = await getExecutionData(data.executionId);
```

Comment:
so, if we are fetching the full execution data at the end, we could update the `executionFinished` push event to not include the data anymore.